stylix/palette.nix

```diff
@@ -1,25 +1,106 @@
-{ palette-generator, base16 }:
 {
+  palette-generator,
+  base16,
+}: {
   pkgs,
   lib,
   config,
   options,
   ...
-}:
-
-let
+}: let
   cfg = config.stylix;
-
-in
-{
+  adjustLightness = rgbColorString: primaryScale: let
+    stripped = builtins.replaceStrings ["rgb(" ")"] ["" ""] rgbColorString;
+    values = builtins.split "," stripped;
+    v1 = builtins.fromJSON (builtins.elemAt values 0);
+    v2 = builtins.fromJSON (builtins.elemAt values 2);
+    v3 = builtins.fromJSON (builtins.elemAt values 4);
+    preLightness = (v1 + v2 + v3) / 3.0;
+    adj = (preLightness / 255.0 * (1.0 - primaryScale) + primaryScale) / preLightness * 255.0;
+    v1adj = lib.max (lib.min (v1 * adj) 255.0) 0.0;
+    v2adj = lib.max (lib.min (v2 * adj) 255.0) 0.0;
+    v3adj = lib.max (lib.min (v3 * adj) 255.0) 0.0;
+    round = x: let
+      floored = builtins.floor x;
+      diff = x - floored;
+    in
+      if diff >= 0.5
+      then floored + 1
+      else floored;
+  in
+    builtins.toString (lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v1adj)))
+    + builtins.toString (lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v2adj)))
+    + builtins.toString (lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v3adj)));
+in {
   options.stylix = {
+    scheme = lib.mkOption {
```

Comment:
Consider grouping all the new options into a subcategory, to make it clear that they're only for generated themes.

Perhaps also move the `polarity` option into that category (unless you're planning to remove it).